src/internal_events/template.rs

```diff
@@ -0,0 +1,30 @@
+use super::InternalEvent;
+use crate::template::TemplateRenderingError;
+use metrics::counter;
+
+pub struct TemplateRenderingFailed<'a> {
+    pub field: Option<&'a str>,
+    pub drop_event: bool,
+    pub error: TemplateRenderingError,
+}
+
+impl<'a> InternalEvent for TemplateRenderingFailed<'a> {
+    fn emit_logs(&self) {
+        let mut msg = "Failed to render template".to_owned();
+        if let Some(field) = self.field {
+            msg.push_str(&format!(" for \"{}\"", field));
+        }
+        if self.drop_event {
+            msg.push_str("; discarding event");
+        }
+        msg.push('.');
+        warn!(message = %msg, error = %self.error, internal_log_rate_secs = 30);
```

Comment:
Thanks to @FungusHumungus , I discovered today that this rate limit actually applies to the call site rather than based on the message or other fields. I think this means, if you had multiple components with template rendering errors, it would be very easy to miss one of them as only one of them would be chosen, per 30s, to output a log message for.

I think we should ideally separate this out so that the rate limit applies per templated option that fails to render but I think this will require some refactoring of the tracing layer that is also blocking #6307

Reply:
This is very unfortunate, I assumed that logs are rate-limited based on the scope of the log. It is not possible to reuse the events if this is the case.

Reply:
Ref #6342